cpp/include/raft/linalg/reduce_cols_by_key.cuh

```diff
@@ -77,14 +79,16 @@ void reduce_cols_by_key(const T* data,
  * @param[out] out the output reduced raft::device_matrix_view along columns (dim = nrows x nkeys).
  * This will be assumed to be in row-major layout
  * @param[in] nkeys number of unique keys in the keys array
+ * @param[in] reset_sums  Whether to reset the output sums to zero before reducing
  */
 template <typename ElementType, typename KeyType = ElementType, typename IndexType = std::uint32_t>
 void reduce_cols_by_key(
   const raft::handle_t& handle,
   raft::device_matrix_view<const ElementType, IndexType, raft::row_major> data,
   raft::device_vector_view<const KeyType, IndexType> keys,
   raft::device_matrix_view<ElementType, IndexType, raft::row_major> out,
-  IndexType nkeys)
+  IndexType nkeys,
```

Comment:
Sure, but what kind of use cases do you have in mind? Since this primitive works with keys in the range `[0, nkeys[` it's fair to assume `nkeys` will typically be known. For a more generic primitive working with arbitrary keys, we would first need to map them to such a range, and we can compute `nkeys` in the process.

Reply:
Unless you mean that instead of passing `nkeys` we should get it from the dimensions of `out`, which actually makes a lot of sense.

Reply:
Yeah, I guess the latter option makes sense. I was considering doing something similar in the knn APIs, since the shape of the output matrices already tell us k.

Reply:
I see the following options, let me know if you have a preference:

- Give `nkeys` a default value and ignore it, using output dims instead (might break compilation due to unused variable warning?)
- Give `nkeys` a default value and if not overridden by the user, get it from the output dims.
- Remove `nkeys` arg (potentially breaking but I doubt anyone is using this prim outside the raft codebase?)

Reply:
The second option looks good to me.

Reply:
@Nyrio do you want me to go ahead and merge this PR in the meantime?